### PR TITLE
Repo setup and v1 of the plugin to intercept superagent requests and …

### PR DESCRIPTION
**Pull request by bdgamble**
…sign them with aws signature v4

### CODE REVIEW

**Comment by bdgamble** [comment]
@dougmoscrop can you review?

**Comment by dougmoscrop** [comment]
Wil do

File: index.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        headers: headers
+      }, {
+        accessKeyId: options.key,
+        secretAccessKey: options.secret,
+        sessionToken: options.sessionToken
+      });
+
+      req.header = signedOptions.headers;
+
+      req.end = end;
```
Comment (dougmoscrop) [inline index.js]: What's the value in putting end 'back' vs. just calling end(cb);

**Comment by dougmoscrop** [comment]
Could be worth adding an integration test against real AWS service.

File: index.js (hunk 1 above)
Comment (bdgamble) [inline index.js]: I had seen  other plugins doing this type of thing done this way and wonder if there could be potential for another plugin that comes after this one that needs access to the original end function on the request

**Comment by bdgamble** [comment]
I'll try to get an integration test in

**Comment by bdgamble** [comment]
@dougmoscrop integration test added which anyone should be able to run

File: serverless.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+package:
+  exclude:
+    - src/**
+    - tests/**
+    - node_modules/**
+  include:
+    - tests/infra/**
+
+custom:
+  resource-output-dir: /tests/infra/resources
```
Comment (dougmoscrop) [inline serverless.yml]: Should this be /tests/ or tests/

File: tests/spec.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        });
+    });
+
+    it('with provided values and adds sessionToken header', () => {
+      return request
+        .post('/')
+        .use(signRequest(service, { key: 'a', secret: 'b', sessionToken: 'c'}))
+        .send(testPostData)
+        .then(resp => {
+          return Promise.all([
```
Comment (dougmoscrop) [inline tests/spec.js]: I don't think you need Promise all here

File: serverless.yml (hunk 2 above)
Comment (bdgamble) [inline serverless.yml]: This was the format the plugin used, and it works, but I can try without leading slash

File: tests/spec.js (hunk 3 above)
Comment (bdgamble) [inline tests/spec.js]: Yah, not sure if I even need to use for any of these, think I got used to writing the tests this way from attributes :P. I'll clean up

**Comment by bdgamble** [comment]
@dougmoscrop should be good now

**Pull request merged**